src/v/kafka/server/group.cc

```diff
@@ -1350,6 +1433,30 @@ group::handle_begin_tx(cluster::begin_group_tx_request&& r) {
     }
 }
 
+ss::future<cluster::prepare_group_tx_reply>
+group::handle_prepare_tx(cluster::prepare_group_tx_request&& r) {
+    if (in_state(group_state::dead)) {
+        cluster::prepare_group_tx_reply reply;
+        reply.ec = cluster::tx_errc::coordinator_not_available;
+        co_return reply;
+    } else if (in_state(group_state::empty)) {
+        co_return co_await prepare_tx(std::move(r));
+    } else if (
+      in_state(group_state::stable)
+      || in_state(group_state::preparing_rebalance)) {
+        co_return co_await prepare_tx(std::move(r));
+    } else if (in_state(group_state::completing_rebalance)) {
+        cluster::prepare_group_tx_reply reply;
+        reply.ec = cluster::tx_errc::rebalance_in_progress;
+        co_return reply;
+    } else {
```

Comment:
these are nits, but the `else` is also not needed here.

Reply:
it follows the pattern set by `handle_leave_group`, and `handle_offset_commit` and I don't want to introduce code which style isn't consistent with the surrounding so leaving as is